debuginfo/src/object.rs

```diff
@@ -379,9 +379,11 @@ pub enum ObjectDebugSession<'d> {
 impl<'d> ObjectDebugSession<'d> {
     /// Returns an iterator over all functions in this debug file.
     ///
-    /// The iteration is guaranteed to be sorted by function address and includes all compilation
```

Comment:
Well, if we would be very pedantic, removing this guarantee would be a breaking change?

Reply:
That guarantee was always broken, this is what this PR is about. I'm thinking since it's broken and the symcache writer already partially handled unsorted functions, we can just remove this here and do a minor bump with a warning notice. 

The next major bump will have changes to the iterators themselves.